backstory/src/main/java/com/google/sps/story/PromptManagerBodyGenerator.java

```diff
@@ -0,0 +1,230 @@
+          "a <adj> <adj> <noun> and a <adj> <adj> <noun> appeared all at once. "));
+
+  /** Templates for introductory sentence completion with gerunds included */
+  private final static List<String> INTRO_TEMPLATES_WITH_GERUNDS = Collections.unmodifiableList(
+      Arrays.asList("there was a <adj> <adj> <noun> <gerund> alongside a <adj> <noun>.",
+          "a <adj> <noun> as well as a <gerund> <adj> <noun> were together."));
+
+  /** Templates for second sentence. */
+  private final static List<String> SECOND_TEMPLATES = Collections.unmodifiableList(
+      Arrays.asList("A <adj> <noun> was also present, quite an interesting scene.",
+          "One must not forget the <adj> <noun>, which simply cannot be ignored."));
+
+  /** Templates for list method sentence endings. */
+  private final static List<String> SIMPLE_ENDINGS =
+      Collections.unmodifiableList(Arrays.asList("were all really quite interesting.",
+          "all came together in one place.", "were all together at once."));
+
+  /** Default ending for empty string return. */
+  private final static String EMPTY_ENDING = "a hectic, unrecognizable scene took place.";
+
+  /**
+   * Initializes fields and takes in word processing client if supplied.
+   *
+   * @param keywords Input words for generation.
+   * @param chooseRandomly Whether or not to randomly select a template.
+   * @param wordTools Client for word processing API calls.
+   */
+  public PromptManagerBodyGenerator(
+      List<String> keywords, boolean chooseRandomly, PromptManagerAPIsClient wordAPIsClient) {
+    this(keywords, chooseRandomly);
+    this.wordAPIsClient = wordAPIsClient;
+  }
+
+  /**
+   * Initializes fields.
+   *
+   * @param keywords Input words for generation.
+   * @param chooseRandomly Whether or not to randomly select a template.
+   */
+  public PromptManagerBodyGenerator(List<String> keywords, boolean chooseRandomly) {
+    this.keywords = keywords;
+    this.chooseRandomly = chooseRandomly;
+    templateChooser = new Random();
+    outputBody = "";
+  }
+
+  /**
+   * Generates prompt body using given labels. If randomness
+   * is disabled, the first of each applicable template type
+   * will be used. Templates are filled using first available
+   * of each WordType.
+   *
+   * @return A String containing the output prompt.
+   */
+  public String generateBody() {
+    try {
+      // Instantiate wordTools within try-catch if not supplied.
+      if (wordAPIsClient == null) {
+        wordAPIsClient = new PromptManagerAPIsClient();
+      }
+
+      // Classify given word inputs.
+      typeMap = wordAPIsClient.groupByWordType(keywords);
+      nouns = typeMap.get(WordType.NOUN);
+
+      // Isolate gerunds
+      gerunds = typeMap.get(WordType.GERUND);
+
+      // Determine method based on given noun count.
+      if (nouns.size() >= MINIMUM_NOUNS) {
+        return makeDescriptiveTemplate();
+      } else {
+        return makeListTemplate();
+      }
+    } catch (Exception e) {
```

Comment:
Can the try/catch be more targeted, so that it's clear what actually might throw? That way if that code is ever removed, we'll know that try/catch can be removed. Also applies to other methods.

Reply:
Done